src/core/ddsc/src/dds_write.c

```diff
@@ -513,22 +550,43 @@ dds_return_t dds_write_impl (dds_writer *wr, const void * data, dds_time_t tstam
   thread_state_awake (ts1, &wr->m_entity.m_domain->gv);
 
   // 3. Check availability of iceoryx and reader status
-
   bool iceoryx_available = wr->m_iox_pub != NULL;
+  //bool has_fixed_size_type = wr->m_topic->m_stype->fixed_size;
+  void *iox_chunk = NULL;
 
   if(iceoryx_available) {
     //note: whether the data was loaned cannot be determined in the non-iceoryx case currently
     if(!deregister_pub_loan(wr, data))
     {
-      void* chunk_data = create_iox_chunk(wr);
-      if(chunk_data) {
-        memcpy (chunk_data, data, wr->m_topic->m_stype->iox_size);
-        data = chunk_data;
+      size_t required_size = get_required_buffer_size(wr->m_topic, data);
```

Comment:
We should fail the operation with `BAD_PARAMETER` if the returned size is `SIZE_MAX` here.

Reply:
Yes, but all these checks which almost always have the same result make me a sad panda. Will add it though.